File: app/src/main/java/com/apkarthik1986/jewelstockmanager/domain/model/BoxConfig.kt
package com.apkarthik1986.jewelstockmanager.domain.model

/**
 * Domain model representing a storage box container.
 * Each box belongs to one jewelry category and holds its own tare weight.
 */
data class BoxConfig(
    val boxNumber: String,
    val category: String,
    val tareWeightGrams: Double,
    val isActive: Boolean = true,
    val location: String = "",
    val lastUpdatedMs: Long = System.currentTimeMillis(),
    val sheetsRowIndex: Int = -1
)

File: lib/models/box_config.dart
class BoxConfig {
  BoxConfig({
    required this.boxNumber,
    required this.category,
    required this.tareWeightGrams,
    required this.location,
    this.isActive = true,
  });

  factory BoxConfig.fromJson(Map<String, dynamic> json) => BoxConfig(
        boxNumber: json['boxNumber'] as String,
        category: json['category'] as String,
        tareWeightGrams: (json['tareWeightGrams'] as num).toDouble(),
        location: json['location'] as String,
        isActive: json['isActive'] as bool? ?? true,
      );

  final String boxNumber;
  final String category;
  final double tareWeightGrams;
  final String location;
  final bool isActive;
}
